java/src/jmri/jmrix/AbstractMRTrafficController.java

```diff
@@ -1094,6 +1098,38 @@ public void run() {
     }  // end XmtNotifier
 
     /**
+     * Terminate the receive and transmit threads.
+     *<p>
+     * This is intended to be used only by testing subclasses.
+     */
+    public void terminateThreads() {
+        threadStopRequest = true;
+        if (xmtThread != null) {
+            xmtThread.stop();
```

Comment:
A flag, without both interrupts and code changes to handle those properly, doesn't end a thread that's waiting for something, e.g. for input.  See comment in main conversation about how this is WIP while those code changes are made.